src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -651,6 +754,20 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     void TermControl::_KeyDownHandler(winrt::Windows::Foundation::IInspectable const& /*sender*/,
                                       Input::KeyRoutedEventArgs const& e)
     {
+        // If Esc is pressed and the search box is opened, we close the search box
+        if (_searchBox && e.OriginalKey() == winrt::Windows::System::VirtualKey::Escape)
+        {
+            _CloseSearchBoxControl(nullptr, nullptr);
+            return;
+        }
+
+        // If the current focused element is a child element of searchbox,
+        // we do not send this event up to terminal
+        if (_searchBox && _searchBox.ContainsFocus())
+        {
+            return;
+        }

```

Comment:
Can this ever be hit? Doesn't all input go to the SearchBox by nature of it having focus?

Reply:
Yes, it will and I think it is important. Terminal control is using a tunneling strategy on key input. Any input (including inputs on the search box because it is a child of TermControl) will be received by TermControl first and handled in this _KeyDownHandler. Thus, I added this check to separate inputs on TermControl and search box, this avoids lots of werid bugs.